agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/BatchItemProcessor.java

```diff
@@ -169,6 +171,23 @@ private void addItem(TelemetryItem item) {
 
     @Override
     public void run() {
+      // incrementing CallDepth for LoggerProvider causes the OpenTelemetry Java agent logging
+      // instrumentation to back off
+      //
+      // note: recursive log capture is only known to be an issue on Wildfly, because it redirects
+      // System.out back to a logging library which is itself instrumented by the Java agent
+      //
+      // see OutOfMemoryWithDebugLevelTest for repro that will fail without this code
+      CallDepth callDepth = CallDepth.forClass(LoggerProvider.class);
```

Comment:
is it necessary to use CallDepth here? Log makes sense. BatchItemProcessor is used for span and metrics too. 
we passed a copy of BatchItemProcessor into AgentLogExporter.

Reply:
this method is run in a different thread

Reply:
can we add it as comment?